### PR TITLE
Disable geocoder api notifications for oss users

### PR DESCRIPTION
**Pull request by ethervoid**
We want to disable and remove noise for the OSS users of CartoDB that don't have, by now, a complete instance of Geocoder API

### CODE REVIEW

**Comment by rafatower** [comment]
as per our conversation, I'd rather prefer to change this:

https://github.com/CartoDB/cartodb/blob/master/app/models/user/db_service.rb#L1332

with that:
```ruby
# If there's no config we assume there's no need to install the geocoder client as it is an independent API
return if config.blank?
```

**Comment by ethervoid** [comment]
It's done in the last commit :|

File: app/models/user/db_service.rb [hunk 1]
```diff
@@ -415,7 +415,14 @@ def install_and_configure_geocoder_api_extension
               SET search_path TO #{build_search_path}") if @user.organization_user?
           return true
         rescue => e
-          CartoDB.notify_error('Error installing and configuring geocoder api extension', error: e.inspect, user: @user)
+          # For now CartoDB.com is the only host with Geocoder API installed
+          # so we want to avoid noise to OSS users
+          if Cartodb.config[:cartodb_com_hosted] == false
```
Comment (rafatower) [inline app/models/user/db_service.rb]: I don't agree with this change

File: config/app_config.yml.sample [hunk 2]
```diff
@@ -171,11 +171,6 @@ defaults: &defaults
       base_url: ''
       api_key: ''
       table_name: ''
-    api:
```
Comment (rafatower) [inline config/app_config.yml.sample]: same must be done in app_config.yml.testing

File: app/models/user/db_service.rb (hunk 1 above)
Comment (ethervoid) [inline app/models/user/db_service.rb]: removed

File: config/app_config.yml.sample (hunk 2 above)
Comment (ethervoid) [inline config/app_config.yml.sample]: This config is not in the testing file

**Comment by ethervoid** [comment]
@rafatower check it to deploy if its ok

**Comment by rafatower** [comment]
Looks ok, but can you please check these errors in the tests?
```
13:43:41 CartoDB::Visualization::TableBlender
13:43:43 error: Error installing and configuring geocoder api extension
13:43:43 {:error=>"#<TypeError: wrong argument type nil (expected String)>", [...]
```
(btw missing a backtrace there, is `inspect` what we want?)

File: config/app_config.yml.sample (hunk 2 above)
Comment (rafatower) [inline config/app_config.yml.sample]: my bad, it was in my PR :dancer:

**Comment by rafatower** [comment]
By looking into the code I'd say we should skip entierly `install_and_configure_geocoder_api_extension` if it is not in app_config

**Comment by ethervoid** [comment]
I come to the same conclusion, context switching isn't good for focus xD

**Comment by ethervoid** [comment]
Solved the error traces. Tomorrow I'll deploy it

**Comment by rafatower** [comment]
:+1: great!

**Pull request merged**